### DISCUSSION TITLE
Resource not accessible by integration - assigning author

### DISCUSSION THREAD

**Discussion by drpaneas** [community/community#25487] [Actions]
<p>I am using <a href="https://github.com/marketplace/actions/auto-author-assign" rel="noopener nofollow ugc">https://github.com/marketplace/actions/auto-author-assign</a><br>
and my workflow is:</p>
<pre><code class="lang-yaml">name: 'Auto Author Assign'
on:
  pull_request:
    types: [opened, reopened]
jobs:
  add-assignees:
    runs-on: ubuntu-latest
    steps:
      - name: auto-author-assign
        uses: toshimaru/auto-author-assign@v1.1.2
        with:
          repo-token: "${{ secrets.GITHUB_TOKEN }}"
</code></pre>
<p>When there is a new PR, the action fails to assign the PR to the person who sent it. See for example: <a href="https://github.com/drpaneas/romie/pull/58/checks?check_run_id=1220186720" rel="noopener nofollow ugc">https://github.com/drpaneas/romie/pull/58/checks?check_run_id=1220186720</a></p>
<p>the error says:</p>
<pre><code class="lang-yaml">Run toshimaru/auto-author-assign@v1.1.2
  with:
    repo-token: ***
Error: Resource not accessible by integration
</code></pre>

**Comment by kingthorin** [comment, marked as answer]
<p>You need to use <code>pull_request_target</code> to allow forks to have access to the token.</p>

**Comment by drpaneas** [comment]
<p>Thanks for your reply. Does that mean I need to create a personal access token?<br>
EDIT: I’ve found this page <a href="https://github.com/actions/checkout/blob/05aa1e5bb412bf00f60524f433ec33689f14e2e5/README.md#Checkout-pull-request-on-pull_request_target" rel="noopener nofollow ugc">https://github.com/actions/checkout/blob/05aa1e5bb412bf00f60524f433ec33689f14e2e5/README.md#Checkout-pull-request-on-pull_request_target</a></p>

**Comment by airtower-luna** [comment]
<p>No, a <code>pull_request</code> workflow from a fork won’t have access to the PAT secret either. A <code>pull_request_target</code> workflow does, and does get a <code>GITHUB_TOKEN</code> with write access. The idea is that the workflow for <code>pull_request_target</code> is already trusted code on the target branch, just make sure not to give any untrusted code from the pull request access to secrets or token.</p>
